cmd/oras/root/manifest/index/create.go

```diff
@@ -188,7 +197,8 @@ func getPlatform(ctx context.Context, target oras.ReadOnlyTarget, manifestBytes
 	return &platform, nil
 }
 
-func pushIndex(ctx context.Context, target oras.Target, desc ocispec.Descriptor, content []byte, ref string, extraRefs []string, path string, printer *output.Printer) error {
+func pushIndex(ctx context.Context, onIndexPushed func(path string) error, onTagged func(desc ocispec.Descriptor, tag string) error,
```

Comment:
This is a private method here, seems like it would make more sense for it to take handlers than function pointers.

Another thing, it seems like an extremely long list of params like something is not right here.

Reply:
Since both `create` and `update` use this method and index create handlers and index update handlers have different types, the function argument can't take handler argument.

Yeah this is an extremely long list of parameters, but I can't find another way :(